src/dgpost/utils/plot.py

```diff
@@ -0,0 +1,126 @@
+        nrows:      !!int       # optional int defining the number of rows of the grid for :class:`matplotlib.gridspec.GridSpec`
+        ncols:      !!int       # optional int defining the number of columns of the grid for :class:`matplotlib.gridspec.GridSpec`
+        save:                   # optional to save the generated plot, `fname` required and any additional kwargs gets passed to :func:`matplotlib.figure.Figure.savefig`
+          fname:    !!str       # path, where the file is saved
+        ax_args:                # sequence of arguments for the different axes
+          cols:     !!tuple     # optional tuple (int, int) given the lower and upper bounds in the grid columns
+          rows:     !!tuple     # optional tuple (int, int) given the lower and upper bounds in the grid rows
+          series:               # sequence of different plot commands
+            y:      !!str       # column label for y data
+            x:      !!str       # optional column label for x data, or else index is used
+            kind:   !!str       # optional kind of plot to produce, default `scatter'
+                                # additional kwargs gets passed to plotting method
+          legend:   !!dict      # TODO
+          methods:  !!dict      # optional kwargs where the key is an attribute of an :class:`matplotlib.axes.Axes`
+                                # and the value is a dict containing kwargs to be called on the selected attribute
+                                # also possible to call method on subobject of the `axes` i.e. `axes.xaxis.set_label_text`
+
+.. codeauthor:: Ueli Sauter
+"""
+import matplotlib
+import pandas as pd
+from matplotlib import pyplot as plt
+from matplotlib.gridspec import GridSpec
+
+
+def apply_plot_style(style: dict):
+    """
+    Updates the plot style with the given dictionary.
+    For available kwargs see :class:`matplotlib.rcParams`.
+    If style is None, applies/resets to the default matplotlib style.
+
+    Parameters
+    ----------
+    style
+        A dictionary object containing valid key/value pairs.
+
+    """
+    if style is not None:
+        matplotlib.rcParams.update(style)
+    else:
+        matplotlib.rcdefaults()
+
+
+def plt_axes(ax: matplotlib.axes.Axes, table: pd.DataFrame, ax_args: dict):
+    """
+    Processes ax_args and plots the data
+
+    Parameters
+    ----------
+    ax
+        axes to be plotted to
+    table
+        dataframe containing the data
+    ax_args
+        arguments for the axes
+
+    """
+    series = ax_args.pop("series", [])
+    for method, arguments in ax_args.pop("methods", {}).items():
+        attr = ax
+        # allows for method calls on attributes
+        for met in method.split("."):
+            attr = getattr(attr, met)
+        attr(**arguments)
+    ax.set(**ax_args)
+    for element in series:
+        y = element.pop("y")
+        y_data = table[y]
```

Comment:
I think we should be accessing the data using `transform.helpers.pQ()`, which guarantees that whatever is in the table will be converted to a `pint.Quantity`.

Reply:
I tried to do it like this before and then decided not to push it because with `pQ` we lose the information about the indices of the `DataFrame` since it gets converted to a `np.array`. 
I did a little workaround, but I don't like it very much. I copied the code from `pQ` and `separate_data` instead of calling the methods. So that I retain the information about the indices without "loading" the data twice.
Do you have an idea? Or how important is the index for the plotting, especially when it gets changed to a multiindex later on?